src/cudamatrix/cu-array-inl.h

```diff
@@ -130,6 +130,23 @@ void CuArray<T>::CopyToVec(std::vector<T> *dst) const {
 
 
 template<typename T>
+void CuArray<T>::CopyToVec(T *dst) const {
```

Comment:
I just realized that CopyToVec is not a very good name because 'dst' is not a Vector, and we need to clarify that 'dst' is on the CPU.  Please change this name to CopyToHost.